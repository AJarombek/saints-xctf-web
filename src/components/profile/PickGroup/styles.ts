/**
 * JSS styles for the PickGroup component.
 * @author Andrew Jarombek
 * @since 12/7/2020
 */

import { FontMixins } from '../../../styles/mixins';
import Colors from '../../../styles/colors';

export default {
  group: {
    display: 'flex',
    alignItems: 'center',
    marginBottom: 10,
    borderRadius: 4,
    boxShadow: 'rgba(170, 170, 170, 0.2) 0px 1px 3px',
    backgroundColor: ({ status }: { status: string }): string =>
      status === 'accepted'
        ? Colors.sxctfRed
        : status === 'pending'
        ? Colors.spotPaletteBrown
        : Colors.lightestBackground,
    color: ({ status }: { status: string }): string => (status ? '#FFF' : '#000')
  },
  groupTitle: {
    ...FontMixins.roboto(),
    marginLeft: 15
  },
  groupActionIcon: {
    margin: '0 15px 0 auto',

    '&:active': {
      backgroundColor: 'transparent !important'
    },

    '& > button > p': {
      ...FontMixins.elegantIcons(),
      color: ({ status }: { status: string }): string => (status ? '#FFF' : '#000')
    }
  }
};
