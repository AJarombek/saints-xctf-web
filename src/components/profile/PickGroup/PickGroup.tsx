/**
 * Component for picking a group to join.
 * @author Andrew Jarombek
 * @since 12/7/2020
 */

import React from 'react';
import { createUseStyles } from 'react-jss';
import styles from './styles';
import { GroupMember } from '../../../redux/types';
import { AJButton } from 'jarombek-react-components';

interface Props {
  group?: GroupMember;
  onMembershipClick: (group: GroupMember) => void;
  joined: boolean;
  left: boolean;
}

const useStyles = createUseStyles(styles);

const PickGroup: React.FunctionComponent<Props> = ({ group, onMembershipClick, joined, left }) => {
  const classes = useStyles({ status: joined ? 'pending' : left ? null : group.status });

  return (
    <div className={classes.group} data-cypress="pickGroup">
      <p className={classes.groupTitle}>{group.group_title}</p>
      <AJButton type="text" className={classes.groupActionIcon} onClick={(): void => onMembershipClick(group)}>
        {(((group.status === 'accepted' || group.status === 'pending') && !left) || joined) && <p>&#x0051;</p>}
        {((group.status === null && !joined) || left) && <p>&#x0050;</p>}
      </AJButton>
    </div>
  );
};

export default PickGroup;
