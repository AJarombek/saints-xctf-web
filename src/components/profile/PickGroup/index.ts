import PickGroup from './PickGroup';

export default PickGroup;
