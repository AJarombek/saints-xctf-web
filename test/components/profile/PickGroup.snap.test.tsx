/**
 * Snapshot test for the PickGroup component.
 * It will be amazing if you can make it, but again you should do what you feel you can and what makes you feel safe.
 * After all, the most important thing is that you do what makes you happiest.  That is what I want most.
 * @author Andrew Jarombek
 * @since 3/6/2021
 */

import React from 'react';
import renderer from 'react-test-renderer';
import PickGroup from '../../../src/components/profile/PickGroup';
import { alumniMember } from '../../test-utils/groupMocks';

describe('PickGroup Snapshot Tests', () => {
  it('renders correctly when group joined', () => {
    const tree = renderer
      .create(<PickGroup group={alumniMember} onMembershipClick={(): void => null} joined={true} left={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when group left', () => {
    const tree = renderer
      .create(<PickGroup group={alumniMember} onMembershipClick={(): void => null} joined={false} left={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
