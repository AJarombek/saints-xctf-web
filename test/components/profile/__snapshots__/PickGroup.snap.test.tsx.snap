// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickGroup Snapshot Tests renders correctly when group joined 1`] = `
<div
  className="group-0-1-1 group-d0-0-1-4"
  data-cypress="pickGroup"
>
  <p
    className="groupTitle-0-1-2"
  >
    Alumni
  </p>
  <div
    className="ajBaseButton-2-2-4 ajBaseButtonEnabled-2-2-5 aj-text-button ajTextButton-2-2-1 ajTextButtonEnabled-2-2-2 groupActionIcon-0-1-3 groupActionIcon-d1-0-1-5"
    onClick={[Function]}
  >
    <button
      disabled={false}
      type="button"
    >
      <p>
        Q
      </p>
    </button>
  </div>
</div>
`;

exports[`PickGroup Snapshot Tests renders correctly when group left 1`] = `
<div
  className="group-0-1-1 group-d2-0-1-6"
  data-cypress="pickGroup"
>
  <p
    className="groupTitle-0-1-2"
  >
    Alumni
  </p>
  <div
    className="ajBaseButton-2-2-4 ajBaseButtonEnabled-2-2-5 aj-text-button ajTextButton-2-2-1 ajTextButtonEnabled-2-2-2 groupActionIcon-0-1-3 groupActionIcon-d3-0-1-7"
    onClick={[Function]}
  >
    <button
      disabled={false}
      type="button"
    >
      <p>
        P
      </p>
    </button>
  </div>
</div>
`;
